cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -398,8 +404,8 @@ public List<ExecutionTask> getInterBrokerReplicaMovementTasks(Map<Integer, Integ
           int sourceBroker = task.proposal().oldLeader().brokerId();
           Set<Integer> destinationBrokers = task.proposal().replicasToAdd().stream().mapToInt(ReplicaPlacementInfo::brokerId)
                                                 .boxed().collect(Collectors.toSet());
-          if (brokerInvolved.contains(sourceBroker)
-              || KafkaCruiseControlUtils.containsAny(brokerInvolved, destinationBrokers)) {
+          if (_preferRoundRobin && (brokerInvolved.contains(sourceBroker)
```

Comment:
No, it can still move more than one partition of the same broker. The outer `while` loop will go through the same process to pick another broker's move tasks, which could involve brokers that have been visited. But there is no guarantee. In practice, in the first few rounds, usually only 1 or 2 tasks are picked for the broker that has the most number of tasks .

Reply:
Ah I see. That makes sense.